src/redux/user/userReducer.js

```diff
@@ -0,0 +1,56 @@
+import { SIGN_UP_USER, LOGIN_USER, LOGOUT_USER } from "../constants";
+
+const initialState = JSON.parse(localStorage.getItem("userLoginData")) || {
```

Comment:
Please name variables properly because the array of `users` has nothing to do with the user's login data.